src/sysusers/sysusers.c

```diff
@@ -955,7 +956,7 @@ static int add_user(Item *i) {
                 r = uid_is_ok(i->uid, i->name);
                 if (r < 0)
                         return log_error_errno(r, "Failed to verify uid " UID_FMT ": %m", i->uid);
-                if (r == 0) {
+                if (r == 0 && !i->force_reuse_id) {
```

Comment:
The check should be moved to `uid_is_ok()`. If the flag is up, then skip to check group related database.

Reply:
Sure, I updated the PR to move this into uid_is_ok(). I added a new flag (instead of a boolean) because it felt easier to read this way. But I'm happy to tweak if you prefer a different approach (boolean or just taking *Item directly). I also added another test-case to ensure that the reuse of existing IDs is correct to ensure that my code change does not break existing behavior. Please let me know if there is anything else I can do :)